sdk/azcore/internal/exported/response_error_test.go

```diff
@@ -473,3 +475,74 @@ func TestNilResponseBody(t *testing.T) {
 	const expected = "Request information not available\n--------------------------------------------------------------------------------\nRESPONSE 0: \nERROR CODE UNAVAILABLE\n--------------------------------------------------------------------------------\nResponse contained no body\n--------------------------------------------------------------------------------\n"
 	require.EqualValues(t, expected, (&ResponseError{RawResponse: &http.Response{}}).Error())
 }
+
+func TestLogResponseErrorCodeInBody(t *testing.T) {
+	fakeURL, err := url.Parse("https://fakeurl.com/the/path?qp=removed")
+	require.NoError(t, err)
+	rawlog := map[log.Event]string{}
```

Comment:
Might be useful to test the number of events also i.e.
```suggestion
	rawlog := map[log.Event][]string{}
```

Reply:
Good idea, would catch inadvertent extra calls to `log.Write`